File: object_relations/app/models/restaurant.rb
class Restaurant
  attr_accessor :name
  attr_reader :customers, :reviews
  @@all = []
  def initialize(name)
    @@all << self 
    @name = name
    @customers = []
    @reviews = []
  end

  def self.all
    @@all 
  end

  def self.find_by_name(name)
    self.all.find{|restaurant| restaurant.name == name }
  end
  
end

File: javascript/models/restaurant.js
class Restaurant{
 constructor(name){
  this.name = name 
  this.reviews = []
  this.customers = []
  store["restaurants"].push(this)
 }

 static all(){
  return store["restaurants"]
 }

 static findByName(name){
    return store["restaurants"].find(function find(restaurant){return restaurant.name == name})
  }

  reviews(){
    return this.reviews
  }

  customers(){
    return this.customers
  }
}